Code/Framework/AzCore/AzCore/Debug/Trace.cpp

```diff
@@ -356,7 +390,7 @@ namespace AZ::Debug
             else
 #endif //AZ_ENABLE_TRACE_ASSERTS
             // Crash the application directly at assert level 3
-            if (currentLevel >= assertLevel_crash)
+            if (currentLevel == assertLevel_crash)
```

Comment:
Note I'm changing this in case we want to reserve assert levels 4 and up for other things in the future (instead of having undocumented behavior here where an assert level higher than 3 causes a crash implicitly)